File: frontend-mobile/lib/models/booking.dart
import 'package_tier.dart';

class Booking {
  final String id;
  final String travelerId;
  final String tourPackageId;
  final String tourPackageName;
  final PackageTier packageTier;
  final int groupSize;
  final String startDate;
  final String endDate;
  final double budgetPerPerson;
  final String status;
  final bool isLargeGroup;

  Booking({
    required this.id,
    required this.travelerId,
    required this.tourPackageId,
    required this.tourPackageName,
    required this.packageTier,
    required this.groupSize,
    required this.startDate,
    required this.endDate,
    required this.budgetPerPerson,
    required this.status,
    required this.isLargeGroup,
  });

  factory Booking.fromJson(Map<String, dynamic> json) => Booking(
        id: json['id'] as String,
        travelerId: json['travelerId'] as String,
        tourPackageId: json['tourPackageId'] as String,
        tourPackageName: json['tourPackageName'] as String,
        packageTier: PackageTier.fromJson(json['packageTier'] as Map<String, dynamic>),
        groupSize: (json['groupSize'] as num).toInt(),
        startDate: json['startDate'] as String,
        endDate: json['endDate'] as String,
        budgetPerPerson: (json['budgetPerPerson'] as num).toDouble(),
        status: json['status'] as String,
        isLargeGroup: json['isLargeGroup'] as bool,
      );
}

File: backend/src/TrailWise.Domain/Entities/Booking.cs
using TrailWise.Domain.Enums;

namespace TrailWise.Domain.Entities;

public class Booking : BaseEntity
{
    public Guid TravelerId { get; set; }
    public User Traveler { get; set; } = null!;

    public Guid TourPackageId { get; set; }
    public TourPackage TourPackage { get; set; } = null!;

    public Guid PackageTierId { get; set; }
    public PackageTier PackageTier { get; set; } = null!;

    public int GroupSize { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public decimal BudgetPerPerson { get; set; }
    public string? SpecialRequests { get; set; }
    public BookingStatus Status { get; set; } = BookingStatus.Requested;

    public ICollection<BookingAddOn> BookingAddOns { get; set; } = new List<BookingAddOn>();
    public ICollection<GuideAvailability> GuideAvailabilities { get; set; } = new List<GuideAvailability>();
    public ICollection<VehicleAssignment> VehicleAssignments { get; set; } = new List<VehicleAssignment>();
    public ICollection<ItineraryStep> ItinerarySteps { get; set; } = new List<ItineraryStep>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
    public ICollection<AgentWorkflowRun> AgentWorkflowRuns { get; set; } = new List<AgentWorkflowRun>();
}
